### PR TITLE
refactor: tm2 clean up against the latest main branch

### PR DESCRIPTION
**Pull request by piux2**
<details><summary>Original text from #825</summary>
<p>

# Description
Part of #692 #585 

BREAKING CHANGE: In GitHub Action Flow, the build script needs to be updated on moving tm2txsync from tm2/cmd to gno.land/cmd

## gno.land/ 

It provides information and tools necessary for running gno.land node. It includes 
- a node application for starting gnoland nodes to deliver an RPC endpoint
- gnoweb for web access to the chain. 
- tools to interact with gno nodes.
- a cosmos SDK gno VM module with keeper and handler.

In the future, we plan to consolidate tools for node deployment and monitoring. The future plan also includes supporting both IBC and ICS.

## gnovm/ 

Gnovm is agnostic to any SDK or chain.

It is intended for gno contract developers and gno VM developers. It includes 
- a gno development CLI
- gnolang language
- virtual machine implementation.
- standard libraries for gno contracts, 
- examples and test cases.

Future plans include making gnovm compatible with IBC

## tm2/ 

Tendermint2 is a BFT consensus engine and a set of tools for blockchain builders

More details can be found at 
[https://github.com/tendermint/tendermint2#readme](https://github.com/tendermint/tendermint2#readme)

In the future, tm2 plans to:

- Remain synchronized with the tendermint2 repository.
- Port issue fixes between Tendermint v0.23.4 and v0.34.
- Ensure compatibility with IBC and ABCI2.

# The dependences among tm2, gnovm and gno.land

- gno.land depends on tm2 and gnovm
- gnovm depends on tm2 
- tm2 is independent to gnovm and gno.land

```mermaid

flowchart LR
    A[gno.land]--> B[gnovm] --> C[tm2]
    A --> C

```

# Discussion

## client commands 

We aim to establish a clear dependency structure between these components as outlined earlier.

Ideally we'd like to have clear dependency between these components as listed above

At present, tm2/pkg/crypto/keys/client/root.go is dependent on gno.land/pkg/sdk/vm. 
We're considering moving the following commands to gno.land/cmd/gnokey/ and eliminating the newMakeTxCmd() from cmd.AddSubCommands() in tm2/pkg/crypto/client/root.go:

- addpkg.go
- call.go
- send.go
- maketx.go 

However, these files rely on the private config structure and private methods defined in tm2/pkg/crypto/keys/client/. Until we decide the next steps, these files will remain in tm2, which means that tm2/ continues to depend on gno.lang/ for now.

## genproto

As for the genproto, we need to decide where misc/genproto/genproto.go should be placed.

## sample tendermint2 program

Since we've moved out applications that depend on gno in tm2, we need to consider whether we should reintroduce the sample tendermint application or create a new one.

[tendermint classic main.go](https://github.com/tendermint/classic/tree/master/cmd/tendermint)

</p>
</details> 

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Review by thehowl — COMMENTED** [review]
Small comments, I'll be closing #825 and redirecting it to this one for clarity.

My 2c on what to do for `pkg/crypto/keys/client`:

- move `HomeDir` to somewhere under `gnovm` - `pkg/util`? `pkg/fsutil`? `pkg/cliutil`?
  (I like `pkg/fsutil`)
- make all the `newXCmd` functions exported, and all `xCfg`.
   Rationale: we want to separate `addpkg` and `call` from the rest, making `client` a generic package to create tendermint 2 clients. This allows us to enable that.
- Make `SignHandler` unexported (unclear to me why it's exported in the first place -- it's a bit useless as well because `signCfg` is unexported).
- Move `addpkg` and `call` to either `gnovm` or `gno.land`.

If you think this is worth discussing more before deciding, we can go ahead with this one as manfred said in #825 and then tackle addpkg/call in another PR.

File: tm2/Makefile
[the remark this answers is not in the document]
Comment (piux2) [reply tm2/Makefile:1]: @thehowl good catch! gno.land/Makefile missed the gnotxsync, updated it.  

Let's merge it first.  

To make newXCmd and xCfg all exported in tm2/pkg/crypto/keys/client is straight forward. However, I also feel the entire tm2/pkg/crypto/keys/client as part of the tool set could be used a fork instead a library. Suddenly, all the private sub xCmd and  xCfg start making sense. 

I will open RFC for the next step.

File: tm2/Makefile [hunk 1]
```diff
@@ -7,18 +7,16 @@ help:
...
-_build.tm2txsync:;  go build -o build/tm2txsync  ./cmd/tm2txsync
 _build.aminoscan:;  go build -o build/aminoscan  ./pkg/amino/cmd/aminoscan
 _build.goscan:;     go build -o build/goscan     ./pkg/amino/cmd/goscan
 _build.logjack:;    go build -o build/logjack    ./pkg/autofile/cmd
 _build.iaviewer:;   go build -o build/iaviewer   ./pkg/iavl/cmd/iaviewer
 
 .PHONY: install
 install:
-	go install ./cmd/...
+#	go install ./cmd/...
```
Comment (thehowl) [inline tm2/Makefile:19]: maybe remove entire install command?

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

I've left a minor comment just for discussion, since the dependency stuck out

**Review by thehowl — APPROVED** [review]

File: tm2/Makefile (hunk 1 above)
Comment (moul) [reply tm2/Makefile:19]: It will make sense soon, we can comment now and reactivate later, or just let as it is if it's not generating errors/warnings.

**Comment by moul** [comment]
### High-level strategy regarding the splits:
- tm2 should never depend on gno.
- gnovm can depend on tm2 even if it would be better to not depend on it at all.
- gno.land depends on both, it's also OK to keep gno.land as a temporary buffer for everything not generic enough with the goal to split later.

### Target
- tm2/*: allow creating tm2 blockchains and CLIs, without gnovm (all the verbs in tm2/vm/sdk, except call and addpkg)
- gnovm: allow creating tm2 blockchains with gnovm (can depend on tm2); in a far future, we can make it completely independent
- gno.land: compose tm2 + gnovm & configure settings

### Potential intermediary state (not perfect, bug good enough, and flexible)
- tm2: things that are obviously already generic enough for tm2/ in the target
- gnovm: only the language, but nothing regarding "tm2 module" yet
- gno.land: everything mixing gnovm and gno.land, with the goal to extract a few things to gnovm later

File: gnovm/Makefile [hunk 2]
```diff
@@ -38,7 +38,7 @@ GOTEST_FLAGS ?= -v -p 1 -timeout=30m
 
 .PHONY: _test.cmd
 _test.cmd:
-	go test ./cmd/... $(GOTEST_FLAGS)
+	go test ./cmd/gno/... $(GOTEST_FLAGS)
```
Comment (moul) [inline gnovm/Makefile:41]:
```suggestion
	go test ./cmd/... $(GOTEST_FLAGS)
```

I suggest keeping `_test.cmd` the all-in-one cmd/ tester command. If needed, we can add subcommands.

**Review by moul — APPROVED** [review]

**Pull request merged**